//
// Copyright (C) 2024 EA group inc.
// Author: Jeff.li lijippy@163.com
// All rights reserved.
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published
// by the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//
//


syntax="proto2";
option cc_generic_services=true;

package melon;

// The request sent to -trackme_server
message TrackMeRequest {
  optional int64 rpc_version = 1;
  optional string server_addr = 2;    
};

enum TrackMeSeverity {
  TrackMeOK = 0;
  TrackMeWarning = 1;
  TrackMeFatal = 2;
};

// The response from -trackme_server
message TrackMeResponse {
  // Print an ERROR log with error_text when severity=TrackMeFatal
  // Print a WARNING log with error_text when severity=TrackMeWarning
  // Do nothing when severity=TrackMeOk
  optional TrackMeSeverity severity = 1; 
  optional string error_text = 2;
  // If this field is set, send trackme requests with this interval.
  optional int32 new_interval = 3;
};

service TrackMeService {
  rpc TrackMe(TrackMeRequest) returns (TrackMeResponse);
}
